test/hotspot/jtreg/compiler/lib/verify/Verify.java

```diff
@@ -202,83 +310,200 @@ private static void checkEQimpl(MemorySegment a, MemorySegment b, String context
+    private void checkEQimpl(double[] a, double[] b, String field, Object aParent, Object bParent) {
+        if (a.length != b.length) {
+            System.err.println("ERROR: Verify.checkEQ failed: length mismatch: " + a.length + " vs " + b.length);
+            print(a, b, field, aParent, bParent);
+            throw new VerifyException("Double array length mismatch.");
+        }
+
+        for (int i = 0; i < a.length; i++) {
+            if (isDoubleEQ(a[i], b[i])) {
+                System.err.println("ERROR: Verify.checkEQ failed: value mismatch at " + i + ": " + a[i] + " vs " + b[i] + ". check raw: " + isFloatCheckWithRawBits);
+                print(a, b, field, aParent, bParent);
+                throw new VerifyException("Double array value mismatch " + a[i] + " vs " + b[i]);
+            }
+        }
+    }
+
+    /**
+     * Verify that the content of two boolean arrays is identical.
+     */
+    private void checkEQimpl(boolean[] a, boolean[] b, String field, Object aParent, Object bParent) {
+        if (a.length != b.length) {
+            System.err.println("ERROR: Verify.checkEQ failed: length mismatch: " + a.length + " vs " + b.length);
+            print(a, b, field, aParent, bParent);
+            throw new VerifyException("Boolean array length mismatch.");
+        }
+
+        for (int i = 0; i < a.length; i++) {
+            if (a[i] != b[i]) {
+                System.err.println("ERROR: Verify.checkEQ failed: value mismatch at " + i + ": " + a[i] + " vs " + b[i]);
+                print(a, b, field, aParent, bParent);
+                throw new VerifyException("Boolean array value mismatch.");
+            }
+        }
     }
 
     /**
      * Verify that the content of two Object arrays is identical, recursively:
      * every element is compared with checkEQimpl for the corresponding type.
      */
-    private static void checkEQimpl(Object[] a, Object[] b, String context) {
+    private void checkEQimpl(Object[] a, Object[] b, String field, Object aParent, Object bParent) {
         // (1) Length mismatch
         if (a.length != b.length) {
             System.err.println("ERROR: Verify.checkEQ failed: length mismatch: " + a.length + " vs " + b.length);
+            print(a, b, field, aParent, bParent);
             throw new VerifyException("Object array length mismatch.");
         }
 
         for (int i = 0; i < a.length; i++) {
             // Recursive checkEQ call.
-            checkEQ(a[i], b[i], "[" + i + "]" + context);
+            checkEQdispatch(a[i], b[i], "[" + i + "]", a, b);
         }
     }
 
-    private static void print(Object a, String context) {
-        if (a == null) {
-            System.err.println("  " + context + ": null");
+    private void checkEQArbitraryClasses(Object a, Object b) {
+        Class c = a.getClass();
+        while (c != Object.class) {
+            for (Field field : c.getDeclaredFields()) {
+                Object va = null;
+                Object vb = null;
```

Comment:
`null` can be omitted:
```suggestion
                Object va;
                Object vb;
```